tests/e2e/e2e_suite_test.go

```
@@ -0,0 +1,208 @@
+	suite.Run(t, new(IntegrationTestSuite))
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up e2e integration test suite...")
+	var err error
+
+	s.loadUpgradeParams()
+
+	s.upgradeManager, err = upgrade.NewManager()
+	s.NoError(err, "upgrade manager creation error")
+
+}
+
+func (s *IntegrationTestSuite) loadUpgradeParams() {
+	preV := os.Getenv("INITIAL_VERSION")
+	if preV == "" {
+		s.Fail("no pre-upgrade version specified")
+	}
+	s.upgradeParams.InitialVersion = preV
+
+	postV := os.Getenv("TARGET_VERSION")
+	if postV == "" {
+		s.Fail("no post-upgrade version specified")
+	}
+	s.upgradeParams.TargetVersion = postV
+
+	migrateGenFlag := os.Getenv("MIGRATE_GENESIS")
+	migrateGenesis, err := strconv.ParseBool(migrateGenFlag)
+	s.Require().NoError(err, "invalid migrate genesis flag")
+	s.upgradeParams.MigrateGenesis = migrateGenesis
+
+	skipFlag := os.Getenv("E2E_SKIP_CLEANUP")
+	skipCleanup, err := strconv.ParseBool(skipFlag)
+	s.Require().NoError(err, "invalid skip cleanup flag")
+	s.upgradeParams.SkipCleanup = skipCleanup
+
+	mountPath := os.Getenv("MOUNT_PATH")
+	if postV == "" {
+		s.Fail("no mount path specified")
+	}
+	s.upgradeParams.MountPath = mountPath
+	s.T().Log("upgrade params: ", s.upgradeParams)
+}
+
+func (s *IntegrationTestSuite) TearDownSuite() {
+	if s.upgradeParams.SkipCleanup {
+		return
+	}
+	s.T().Log("tearing down e2e integration test suite...")
+
+	s.Require().NoError(s.upgradeManager.KillCurrentNode())
+
+	s.Require().NoError(s.upgradeManager.RemoveNetwork())
+
+	s.Require().NoError(os.RemoveAll(strings.Split(s.upgradeParams.MountPath, ":")[0]))
+}
+
+func (s *IntegrationTestSuite) runInitialNode() {
+	node := upgrade.NewNode(localRepository, s.upgradeParams.InitialVersion)
+	err := s.upgradeManager.RunNode(node)
+	s.NoError(err, "can't run initial node")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	err = s.upgradeManager.WaitForHeight(ctx, 5)
+	s.NoError(err)
```

```suggestion
	s.Require().NoError(err)
```

---

I thought that direct `NoError()` is a shortcut for `Require().NoError()`

According to docs `NoError()` can be used
https://github.com/stretchr/testify/blob/v1.8.1/suite/suite.go#L56

---

correct, but that is an assert, not require. We use require so that the tests fail early